src/qthotkey/plugin.cc

```diff
@@ -436,13 +440,27 @@ bool GlobalHotkeys::init()
 {
     audqt::init();
 
-    if (!QX11Info::isPlatformX11())
+#ifndef Q_OS_UNIX
+    bool   native = false;
+#elif QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+    auto * native = qApp->nativeInterface<QNativeInterface::QX11Application>();
+#else
+    bool   native = QX11Info::isPlatformX11();
+#endif
```

Comment:
I would write it like that, without the indentation spaces and a properly named `bool` for all cases.

```c++
#ifndef Q_OS_UNIX
    bool has_x11 = false;
#elif QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
    QNativeInterface::QX11Application * x11_interface =
        qApp->nativeInterface<QNativeInterface::QX11Application>();
    bool has_x11 = x11_interface != nullptr;
#else
    bool has_x11 = QX11Info::isPlatformX11();
#endif
```

Reply:
Yea, I was thinking about how badly I formatted that piece of code, took your version.